File: software/examples/vbnet/ExampleSimple.vb
Imports System
Imports System.Threading
Imports Tinkerforge

Module ExampleSimple
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Industrial Digital Out 4 Bricklet 2.0

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim ido4 As New BrickletIndustrialDigitalOut4V2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Set channels alternating high/low 10 times with 100 ms delay
        Dim i As Integer
        For i = 0 To 9
            Thread.Sleep(100)
            ido4.SetValue(new Boolean(){True, False, False, False})
            Thread.Sleep(100)
            ido4.SetValue(new Boolean(){False, True, False, False})
            Thread.Sleep(100)
            ido4.SetValue(new Boolean(){False, False, True, False})
            Thread.Sleep(100)
            ido4.SetValue(new Boolean(){False, False, False, True})
        Next i

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/industrial_digital_out_4_v2_bricklet"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
	"time"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Industrial Digital Out 4 Bricklet 2.0.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	ido4, _ := industrial_digital_out_4_v2_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Set channels alternating high/low 10 times with 100 ms delay
	for i := 0; i < 10; i++ {
		time.Sleep(100 * time.Millisecond)
		ido4.SetValue([4]bool{true, false, false, false})
		time.Sleep(100 * time.Millisecond)
		ido4.SetValue([4]bool{false, true, false, false})
		time.Sleep(100 * time.Millisecond)
		ido4.SetValue([4]bool{false, false, true, false})
		time.Sleep(100 * time.Millisecond)
		ido4.SetValue([4]bool{false, false, false, true})
	}

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
